website/docs/reference/appsmith-architecture/README.md

```diff
@@ -0,0 +1,59 @@
+    3. An action execution module that executes your queries to your databases and APIs
+    4. A git module that maintains clones of your Git-connected apps on the file system.
+
+2. The **NGINX server**, which routes:
+    1. Requests to static assets like Javascript, CSS, and images
+    2. Incoming requests to the backend server, or the RTS depending on the request path
+    3. Path-unidentified requests to an `index.html` page with a 200 status code. This is like a typical single-page application where the client React code is expected to understand the path and show the UI accordingly.
+
+3. The **Real Time Service (RTS)**, a tiny NodeJS server that handles a few features such as mapping dependencies between APIs, queries, and web UI components. It also updates references for all reference-able entities when their names change in Appsmith.
+
+4. The **MongoDB server**, which stores all persistent data needed and generated by Appsmith’s building blocks, like:
+    1. Your users’ details and permissions
+    2. Workspaces you have created
+    3. Applications you are building and have deployed
+    4. Datasources you have connected to
+    5. Queries that connect those datasources to your apps.
+
+5. The **Redis server**, which is used for storing user sessions and cache permissions that a user has access to directly and indirectly.
+
+## Process management
+
+To manage the processes needed to function, Appsmith uses [Supervisor](http://supervisord.org/). This is a lightweight process manager that, when given a list of things to run, runs them, restarts them if they crash, and collects their logs for maintenance and debugging.
+
+When you run the Docker command to start the Appsmith container, it identifies which processes are required to start and run. This includes the backend server, the RTS, and the NGINX server -- as well as any additional processes that haven't been plugged in by the user externally, such as a custom MongoDB or Redis instance.
+
+For each of these processes, the container’s entry-point script (the one that Docker invokes when the container starts) generates a config file with all the details needed for supervisord. For example, the config file generated for the backend Java server includes the full set of commands for starting the process, as well as all necessary environment variables and settings. This same entry-point script also helps set up SSL certificates and install custom CA roots.
+
+Supervisord also comes with a command line tool that you can use to control these processes individually. For example, you can run `supervisorctl restart redis` if you want to restart the Redis process. Or, as another example, the `Save and Restart` button under `Admin Settings` on the Appsmith UI is powered the `supervisorctl` tool; clicking that button after you make changes with the UI prompts the backend server to execute `supervisorctl restart backend`, which restarts the backend process.
+
+### Docker
```

Comment:
Removed most of the mentioned Docker and Kubernetes subheadings -- you're right, they were instructing the user on deploying their own instance rather than talking about how Appsmith itself is structured. The information in these sections already exists on other pages, so it was removed.

I combined a bit from each of those into a new heading **Deployment** that informs the user how the Appsmith Cloud is deployed.

Reply:
Do we need to talk about deployment under Architecture?

Reply:
IMO, the content under Deployment head can be a part of How it works itself. Thoughts??

Reply:
@mohanarpit - Could you please suggest an alternative here?